src/Symfony/Component/VarDumper/Caster/Caster.php

```diff
@@ -54,7 +54,7 @@ public static function castObject($obj, \ReflectionClass $reflector)
         if ($a) {
             $p = array_keys($a);
             foreach ($p as $i => $k) {
-                if (isset($k[0]) && "\0" !== $k[0] && !$reflector->hasProperty($k)) {
+                if (isset($k[0]) ? "\0" !== $k[0] && !$reflector->hasProperty($k) : \PHP_VERSION_ID >= 70200) {
```

Comment:
@nicolas-grekas is there any need of writing such agly assertions? ;) I think some brackets would help in reading this code